examples/vision/semisupervised_simclr.py

```diff
@@ -64,204 +65,211 @@
+        .prefetch(AUTOTUNE)
     )
     test_dataset = (
-        tfds.load("stl10", split="test", as_supervised=True)
-        .batch(batch_size)
-        .prefetch(buffer_size=tf.data.AUTOTUNE)
+        tfds.load("stl10", data_dir="dataset", split="test", as_supervised=True)
+        .batch(TEST_BATCH_SIZE, num_parallel_calls=AUTOTUNE)
+        .prefetch(AUTOTUNE)
     )
 
-    # Labeled and unlabeled datasets are zipped together
-    train_dataset = tf.data.Dataset.zip(
-        (unlabeled_train_dataset, labeled_train_dataset)
-    ).prefetch(buffer_size=tf.data.AUTOTUNE)
-
-    return train_dataset, labeled_train_dataset, test_dataset
+    return unlabeled_train_dataset, labeled_train_dataset, test_dataset
 
 
 # Load STL10 dataset
-train_dataset, labeled_train_dataset, test_dataset = prepare_dataset()
+unlabeled_train_dataset, labeled_train_dataset, test_dataset = prepare_dataset()
 
 """
 ## Image augmentations
 
 The two most important image augmentations for contrastive learning are the
 following:
 
-- Cropping: forces the model to encode different parts of the same image
-similarly, we implement it with the
-[RandomTranslation](https://keras.io/api/layers/preprocessing_layers/image_preprocessing/random_translation/)
-and
-[RandomZoom](https://keras.io/api/layers/preprocessing_layers/image_preprocessing/random_zoom/)
-layers
-- Color jitter: prevents a trivial color histogram-based solution to the task by
+- **Cropping**: forces the model to encode different parts of the same image
+similarly.
+- **Color jitter**: prevents a trivial color histogram-based solution to the task by
 distorting color histograms. A principled way to implement that is by affine
 transformations in color space.
 
-In this example we use random horizontal flips as well. Stronger augmentations
-are applied for contrastive learning, along with weaker ones for supervised
-classification to avoid overfitting on the few labeled examples.
+Stronger augmentations are applied for contrastive learning, along with weaker
+ones for supervised classification to avoid overfitting on the few labeled examples.
 
-We implement random color jitter as a custom preprocessing layer. Using
-preprocessing layers for data augmentation has the following two advantages:
-
-- The data augmentation will run on GPU in batches, so the training will not be
-bottlenecked by the data pipeline in environments with constrained CPU
-resources (such as a Colab Notebook, or a personal machine)
-- Deployment is easier as the data preprocessing pipeline is encapsulated in the
-model, and does not have to be reimplemented when deploying it
+We implement the augmentations using the KerasCV library.
 """
 
 
-# Distorts the color distibutions of images
-class RandomColorAffine(layers.Layer):
-    def __init__(self, brightness=0, jitter=0, **kwargs):
-        super().__init__(**kwargs)
-
-        self.brightness = brightness
-        self.jitter = jitter
+def get_augmenter(
```

Comment:
FWIW, `keras_cv` does include a `SimCLRAugmenter` which does this. For the explicitness of the demo, I think it's best to explicitly define it the way you have, but I figured I'd mention it in case you don't know 👍